sklearn_numba_dpex/common/random.py

```diff
@@ -158,39 +158,46 @@ def create_xoroshiro128pp_states(n_states, subsequence_start=0, seed=None, devic
     if hasattr(seed, "randint"):
```

Comment:
NB: diff is hardly readable but the changes are simple, it consists in moving some block of codes into a separate `_make_...` function that is decorated with `lru_cache`.